### PR TITLE
Weld 2.4.7.Final

### PR DESCRIPTION
**Pull request by manovotn**

### CODE REVIEW

**Comment by manovotn** [comment]
@mkouba please review and, if satisfied, publish

File: news/2018-03-20-weld-247Final.asciidoc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+*** Correct proxy generation for class hierarchy with abstract class and generics (link:https://issues.jboss.org/browse/WELD-2470[WELD-2470, window="_blank"])
+** Other issues:
+*** Minor optimization in internal structures of `InterceptionModel` (link:https://issues.jboss.org/browse/WELD-2455[WELD-2455, window="_blank"])
+*** Improve exception logging for `DeploymentException` (link:https://issues.jboss.org/browse/WELD-2453[WELD-2453, window="_blank"])
+*** Prevent NPE if `ProtectionDomain.getPermissions()` returns `null` (link:https://issues.jboss.org/browse/WELD-2464[WELD-2464, window="_blank"])
+
+* Weld SE
+** Fields in `org.jboss.weld.environment.se.Weld` are now `protected` (link:https://issues.jboss.org/browse/WELD-2451[WELD-2451, window="_blank"])
+** Allow to register custom `BeanArchiveHandler` using `ServiceLoader` mechanism (link:https://issues.jboss.org/browse/WELD-2450[WELD-2450, window="_blank"])
+*** This applies to SE and Serblet environments
```
Comment (mkouba) [inline news/2018-03-20-weld-247Final.asciidoc]: Serblet -> Servlet

**Comment by manovotn** [comment]
Fixed and rebased

**Comment by mkouba** [comment]
Published...

**Pull request merged**